pkg/addonmanager/controllers/agentdeploy/controller.go

```diff
@@ -381,6 +381,12 @@ func (c *addonDeployController) buildDeployManifestWorksFunc(addonWorkBuilder *a
 			return nil, nil, fmt.Errorf("failed to get agentAddon")
 		}
 
+		if agentAddon.GetAgentAddonOptions().ConfigCheckEnabled &&
+			!meta.IsStatusConditionTrue(addon.Status.Conditions, addonapiv1alpha1.ManagedClusterAddOnConditionConfigured) {
+			klog.InfoS("Addon configured condition is not set in status", "addonName", addon.Name)
+			return nil, nil, fmt.Errorf("addon %s configured condition is not is not set in status", addon.Name)
```

Comment:
do we need to return error here? I think we do not, the update of addon will retrigger the reconcile.

Reply:
fixed.